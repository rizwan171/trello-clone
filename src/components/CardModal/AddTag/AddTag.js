import React, { useRef, useState } from "react";
import { FiPlus, FiX } from "react-icons/fi";
import { useDispatch, useSelector } from "react-redux";
import Select from "react-select";
import { createTag } from "../../../features/tagsSlice";
import CreateTagMenu from "../TagsMenu/CreateTagMenu/CreateTagMenu";
import TagsMenu from "../TagsMenu/TagsMenu";

const AddTag = ({ card }) => {
  const dispatch = useDispatch();
  const tags = useSelector((state) => state.tags.value);
  const [tagsMenuOpen, setTagsMenuOpen] = useState(false);
  const [createTagOpen, setCreateTagOpen] = useState(false);

  const closeTags = () => {
    setTagsMenuOpen(false);
    setCreateTagOpen(false);
  };

  const showTagsMenu = () => {
    setTagsMenuOpen(true);
    setCreateTagOpen(false);
  };

  const showCreateTagForm = () => {
    setCreateTagOpen(true);
    setTagsMenuOpen(false);
  };

  return (
    <div className="p-1 rounded-ibsm flex items-center cursor-pointer bg-black bg-opacity-5 hover:bg-opacity-10">
      <FiPlus className="mr-0.5 text-trello-gray-200" size={25} onClick={showTagsMenu} />
      {tagsMenuOpen && <TagsMenu card={card} closeTags={closeTags} showCreateTagForm={showCreateTagForm} />}
      {createTagOpen && <CreateTagMenu card={card} closeTags={closeTags} showTagsMenu={showTagsMenu} />}
    </div>
  );
};

export default AddTag;
