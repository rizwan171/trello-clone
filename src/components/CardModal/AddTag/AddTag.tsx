import { FiPlus } from "react-icons/fi";
import { useAppDispatch } from "../../../app/hooks";
import { showTagsMenu } from "../../../features/modalActionMenusVisibilitySlice";

const AddTag = () => {
  const dispatch = useAppDispatch();

  const handleShowTagsMenu = () => {
    dispatch(showTagsMenu());
  };

  return (
    <button
      className="p-1 rounded-ibsm flex items-center cursor-pointer bg-black bg-opacity-5 hover:bg-opacity-10"
      onClick={handleShowTagsMenu}
    >
      <FiPlus className="mr-0.5 text-trello-gray-200" size={25} />
    </button>
  );
};

export default AddTag;
