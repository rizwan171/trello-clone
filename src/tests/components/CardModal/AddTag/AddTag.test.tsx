import { fireEvent, screen } from "@testing-library/react";
import AddTag from "../../../../components/CardModal/AddTag/AddTag";
import { renderWithProviders } from "../../../utils/renderUtils";

describe("AddTag", () => {
  it("should render successfully", () => {
    const view = renderWithProviders(<AddTag />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should show tags menu", () => {
    const { store } = renderWithProviders(<AddTag />);
    const addTagButton = screen.getByRole("button") as HTMLButtonElement;
    expect(addTagButton).toBeInTheDocument();

    const stateBeforeButtonClick = store.getState();
    expect(stateBeforeButtonClick.modalActionMenusVisibility.value.tagsMenuOpen).toBeFalsy();

    fireEvent.click(addTagButton);

    const stateAfterButtonClick = store.getState();
    expect(stateAfterButtonClick.modalActionMenusVisibility.value.tagsMenuOpen).toBeTruthy();
  });
});
