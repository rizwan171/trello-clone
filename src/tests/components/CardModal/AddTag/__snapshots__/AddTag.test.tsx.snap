// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTag should render successfully 1`] = `
<DocumentFragment>
  <button
    class="p-1 rounded-ibsm flex items-center cursor-pointer bg-black bg-opacity-5 hover:bg-opacity-10"
  >
    <svg
      class="mr-0.5 text-trello-gray-200"
      fill="none"
      height="25"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="5"
        y2="19"
      />
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
</DocumentFragment>
`;
